### PR TITLE
Compute expectedCount along traversing pods

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
In DisruptionController#getExpectedScale, the expectedCount can be computed when we traverse the pods and finders.

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by tedyu** [comment]
cc @mattjmcnaughton

File: pkg/controller/disruption/disruption.go [hunk 1]
```diff
@@ -651,6 +652,7 @@ func (dc *DisruptionController) getExpectedScale(pdb *policy.PodDisruptionBudget
 				return
 			}
 			if controllerNScale != nil {
+				expectedCount += controllerNScale.scale - controllerScale[controllerNScale.UID]
```
Comment (mattjmcnaughton) [inline pkg/controller/disruption/disruption.go]: Given that we set `expectedCount` to 0, I'm not sure why we need to include the subtraction here?

Wouldn't `expectedCount += controller.NScale.scale` match the behavior we're removing.

File: pkg/controller/disruption/disruption.go [hunk 2]
```diff
@@ -627,8 +627,9 @@ func (dc *DisruptionController) getExpectedScale(pdb *policy.PodDisruptionBudget
 	// A mapping from controllers to their scale.
 	controllerScale := map[types.UID]int32{}
 
-	// 1. Find the controller for each pod.  If any pod has 0 controllers,
+	// Find the controller for each pod.  If any pod has 0 controllers,
 	// that's an error. With ControllerRef, a pod can only have 1 controller.
+	expectedCount = 0
```
Comment (mattjmcnaughton) [inline pkg/controller/disruption/disruption.go]: Nit: Unrelated to this diff, but is there a reason this function uses named return variables? Imho, it makes the code less readable. If we do move forward with this diff, could I trouble you to refactor to just use a normal return?

**Review by mattjmcnaughton — COMMENTED** [review]
Thanks for your diff!

To be totally honest, I'm not sure I'm in favor of this diff... the optimization doesn't change the asymptotic runtime, and I think it is more readable to have summing all the `expectedCount` as a separate computation at the end (I'd even argue that maybe it should be a separate function...).

Curious to hear the thoughts of others :) At the end of the day, I don't feel super strongly. This is just my initial opinion.

**Comment by tedyu** [comment]
My thinking was that the number of pods may be not small, hence the folding of the count.

Given Matt's feedback, I am willing to shelve this for now.

**Pull request closed without merging**